### ISSUE TITLE
Server Not Accessible Through Public Browser

### ISSUE THREAD

**Issue by iBELLinc** [SkyrimTogether/issues-server#21]
**## Description**

External network clients are not able to access my hosted server via the public browser but they can access via the direct connect menu.

**## Steps to reproduce

How to reproduce this issue:**

1. Run file start.bat in the server directory (can also do this as administrator)
2. External client locates my listing the public servers browser and clicks the _join_ button
3. In-game chat box says it is attempting to establish connection
4. Chat box informs that the connection failed due to an error.
5. Chatbox counts down 4 more attempts before giving up.
6. Client then types in my externalip:port or just externalip and is able to direct connect.

**## Reproduction rate**

This is always the result. I have had 3 different people try and all had the same result. They are all able to access other servers with no issue.

**## Expected result**

Since they can connect via direct connect I would assume connection through the public server browser would also work.

**## Your environment**

*** Launcher version:** Harbor 1.0.6 (myself and all who tested for me)

*** Game edition (choose on which edition do you have problems):**
    * The Elder Scrolls V: Skyrim Special Edition

*** Operating system (choose one):**
    * Windows 10 x64

**## Evidence (optional)**

- I have followed all instructions in this file: https://docs.google.com/document/d/1zQKsZSSwDIe9zOuYRux0_BKXr6ojzXpYFsiYSk-ygxg/edit
and read through the quick fixes document. Nothing worked.

- I attempted disabling the host computer's firewall and antivirus as well as all security setting on the host router. It makes no difference.

- I have a Minecraft server that was port forwarded and I have no issues with external connections on that. Though I noticed that those connections are more like a "direct connect".

- Port checkers can see the Minecraft port is open when the server is running but they indicate that port 10578 is closed regardless of whether server.exe is running or not.

- I attempted using port 12578 instead (properly port forwarding) but this did not work.

- I attempted to use port 25565 (since Minecraft definitely works on this port) but continued to experience the same issue.

- I can ping my server from external networks and it can ping other computers on external networks so communication is open

- I use the DNS service No-IP but have confirmed that using my true external IP address gives the same results. No-IP has also confirmed that they only redirect traffic and it should not effect anything related to what is going on. I can provide their email at request.

- I used Wireshark to track packets but my knowledge is limited on how to read them. It appears that the computer itself or the server software is not responding to communications from external clients. It sends an ACK for the ICMP packets but never sends  other packets to actually establish the connection with the client. I have logs saved so just ask me and I can provide specific screenshots of whatever you want.

- In-game the error sometimes appears as a _time-out error_ for one of the testers but the other two provided screenshots of the message "Connection failed due to an error."

- When sending packets from an external network the external router firewall log was reporting that some packets were "invalid ip packets". Research indicated that perhaps the packets are damaged?

- I have worked on this issue with my ISP and a tech guru who specializes in databases but has networking knowledge. Both have indicated that something must be wrong with the server software or a setting on the computer. After the guru analyzed computer settings including in the registry, processes, and system logs they believe it is most likely an issue with the server software not my computer.

- I have found no other information about a similar issue from anyone else on the internet. The exact opposite issue is reported commonly (can connect publicly but not directly) but this is usually resolved with firewall or antivirus settings. These solutions do not fix the issue I am experiencing.

- Running the server.exe file or server.bat file as admin results in localhost experiencing the same issue as external clients. When the server is not run as administrator then localhost is able to connect via the public server browser.

- I turned on NAT loopback for my Minecraft server because my host computer is also my gaming rig so I host and play on the same machine. This is the command I entered to do that: _CheckNetIsolation.exe LoopbackExempt –a –p=S-1-15-2-1958404141-86561845-1752920682-3514627264-368642714-62675701-733520436_

**My network setup:**

We have two range extenders on the network. The host computer is on one of those extenders but utilizes the same network NOT a subnet. Only one other device is connected to that same extender wirelessly. The host computer is hardwired to that extender and the access point is directly below the actual router (one floor below). Broadcast settings on the extender are hidden because I only want the host computer and this one other device on it to reduce latency to/from the host machine. I never interact with the second extender but it is NOT a subnet as well.

The extenders are both TP-Link RE650 and RE450 respectively. The RE650 is what the host machine utilizes for internet access (no wireless card). There are no options in the extender admin panel to port forward on the extender. The extender has it's own internal ip address. As previously stated: communications between external computers and the host machine work just fine.

**If you would like to try for yourself you are more than welcome to do so. You can connect to my server with the following:**

_legendarc.ddns.net:10578_
or simply _legendarc.ddns.net_

or you can connect to me via the server browser in-game just look for:

_LegendArc | 24/7 | discordlink | All Welcome!_

Happy to provide anything else you might want. Just ask. I have been troubleshooting this for the better part of this past month so I am out of ideas on where to go. I originally assumed this was a router or computer issue which is why I waited so long to post but I am doubting that now since this has stumped so many people.

**Comment by barrosGuilhermeNascimento** [comment]
Have u fixed the issue?

**Comment by iBELLinc** [comment]
@barrosGuilhermeNascimento I was never able to resolve this issue. I reached out twice on the Skyrim Together Subreddit but never got a real response on the second post.
https://www.reddit.com/r/SkyrimTogether/comments/kg2xu4/server_listed_but_not_accessible_through_public/
https://www.reddit.com/r/SkyrimTogether/comments/max7lr/forcing_skyrim_together_to_use_ipv6/

I also posted on r/SysAdmin but was ignored and then the post was marked as spam apparently.

My final thought, as seen in the second post on r/SkyrimTogether was that the software was selecting IPv6 as it's preferred protocol which is not my fault. If there was a way to force the software to bind to IPv4 I think it would fix the issue but I was not able to find a way to do that.

Are you also experiencing this issue?

**Comment by barrosGuilhermeNascimento** [comment]
@iBELLinc
Yes i have the same problem, i was trying to open my own dedicated server to play with a few friends and could make it run outside my network, gonna try the things you said in the second post. I'm trying to take a look too in the new project they're creating that is open source, and letting the user choose the desired ip addres to host the server

**Comment by iBELLinc** [comment]
Do you know if your computer is defaulting to an IPv6 address? If you are unsure you can check here: https://list.skyrim-together.com/list

**Comment by barrosGuilhermeNascimento** [comment]
Mine is binding to my Ipv4 
![image](https://user-images.githubusercontent.com/62022520/128400138-30fc36e1-d0b4-4b7b-891c-3b8da2776757.png)

**Comment by iBELLinc** [comment]
Now that is strange. If you are hosting on IPv4 you should be totally fine. What have you tried so far to troubleshoot? I would assume this is a firewall issue with the computer or router on the host machine.
